tests/Controller/CRUDControllerTest.php

```diff
@@ -1095,11 +1113,14 @@ public function testDeleteActionAjaxSuccess2(): void
 
         $this->request->headers->set('X-Requested-With', 'XMLHttpRequest');
 
+        Request::enableHttpMethodParameterOverride();
+
```

Comment:
```suggestion
```
This should not be enabled in tests.

Reply:
What do you propose to cover the expected behavior then?
Additionally, please note that the original value for `Request::$httpMethodParameterOverride` is being restored after each individual test:
https://github.com/sonata-project/SonataAdminBundle/blob/39c4e2483b956e537b23d42b5491bd036f94c7dc/tests/Controller/CRUDControllerTest.php#L376-L386

Reply:
This should be done in Action, otherwise test will work and action do not:
https://github.com/sonata-project/SonataAdminBundle/blob/39c4e2483b956e537b23d42b5491bd036f94c7dc/src/Controller/CRUDController.php#L220

Reply:
I'm sorry @wbloszyk, I don't understand your suggestion or why you say it won't work.
AFAIK, the choice about enabling or disabling the http method overriding is a global responsibility in the context of the whole application execution. By instance, the configuration node [`framework.http_method_override`](
https://symfony.com/doc/current/reference/configuration/framework.html#http-method-override) establishes this setting globally.

Reply:
https://symfony.com/doc/2.6/book/forms.html#book-forms-changing-action-and-method

> NOTE
If the form’s method is not GET or POST, but PUT, PATCH or DELETE, Symfony will insert a hidden field with the name _method that stores this method. The form will be submitted in a normal POST request, but Symfony’s router is capable of detecting the _method parameter and will interpret it as a PUT, PATCH or DELETE request. Read the cookbook chapter “How to Use HTTP Methods beyond GET and POST in Routes” for more information.

When $disableHttpMethodParameterOverride === false then 
```php
if (Request::METHOD_DELETE === $request->getMethod()) { } // always false
if (Request::METHOD_DELETE === $request->getRestMethod()) { } // always false, before this change - true
```

Reply:
Otherwise `Request::enableHttpMethodParameterOverride()` will be not nesessery in tests.

Reply:
>https://symfony.com/doc/2.6/book/forms.html#book-forms-changing-action-and-method

I'm not sure I understand what are you trying to expose, but I think we should start from the fact that `Request::$httpMethodParameterOverride` is a property that enables the `Request::getMethod()` to return other values than the real HTTP method (`$_SERVER['REQUEST_METHOD']`), based on the "_method" param provided in the `$_POST` or `$_GET` superglobals or in the "X-HTTP-METHOD-OVERRIDE" header.
Regarding this doc entry, be aware that `FormConfigBuilder::setMethod()` is setting this "_method" parameter.

>Otherwise Request::enableHttpMethodParameterOverride() will be not nesessery in tests.

If you run these tests, you'll see that calling `Request::enableHttpMethodParameterOverride()` is necessary to achieve the expected result.

Maybe I'm not understanding the scenario you are talking about.
If that is the case or there is something else I'm omitting or misunderstanding about your concern, please try to share a test case or a alternative PR with what you think should be done. That way I think we could get a clearer perspective.

Reply:
> If you run these tests, you'll see that calling `Request::enableHttpMethodParameterOverride()` is necessary to achieve the expected result.

So code is working when `Request::$httpMethodParameterOverride` is enabled and don't when it is disabled. It is what I mean. I think current code is working correct and we should consider #6046 one more time.